From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ondrej Jirman <megi@xff.cz>
Date: Fri, 24 Jul 2026 19:33:08 +0200
Subject: ARM: dts: sun5i: Wire up USB OTG DMA

Describe the dedicated DMA channel the USB controller's DRQ line is
wired to (DRQ 4 both ways) so the musb glue can use the SoC DMA engine,
and switch to the new A13 compatible that declares this wiring.

Signed-off-by: Ondrej Jirman <megi@xff.cz>
---
 arch/arm/boot/dts/allwinner/sun5i.dtsi | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/arch/arm/boot/dts/allwinner/sun5i.dtsi b/arch/arm/boot/dts/allwinner/sun5i.dtsi
index 111111111111..222222222222 100644
--- a/arch/arm/boot/dts/allwinner/sun5i.dtsi
+++ b/arch/arm/boot/dts/allwinner/sun5i.dtsi
@@ -369,11 +369,14 @@ mmc2: mmc@1c11000 {
 		};
 
 		usb_otg: usb@1c13000 {
-			compatible = "allwinner,sun4i-a10-musb";
+			compatible = "allwinner,sun5i-a13-musb",
+				     "allwinner,sun4i-a10-musb";
 			reg = <0x01c13000 0x0400>;
 			clocks = <&ccu CLK_AHB_OTG>;
 			interrupts = <38>;
 			interrupt-names = "mc";
+			dmas = <&dma SUN4I_DMA_DEDICATED 4>;
+			dma-names = "rxtx";
 			phys = <&usbphy 0>;
 			phy-names = "usb";
 			extcon = <&usbphy 0>;
-- 
Armbian

